kctf-operator/pkg/controller/challenge/service/service.go

```diff
@@ -141,10 +141,8 @@ func generateLoadBalancerService(domainName string, challenge *kctfv1alpha1.Chal
 		service.Spec.Ports = append(service.Spec.Ports, servicePort)
 	}
 
-	if challenge.Spec.Network.Dns == true {
-		service.ObjectMeta.Annotations =
-			map[string]string{"external-dns.alpha.kubernetes.io/hostname": challenge.Name + "." + domainName}
-	}
+	service.ObjectMeta.Annotations =
```

Comment:
I don't feel strongly either way. Technically, we could skip the annotation if we don't have a domain but OTOH it doesn't hurt since the annotation is read by the dns daemon which should handle the "no domain" case.
I think I'm leaning to not having the extra if clause, but if you prefer I should add it then I will :)

Reply:
I mean, does it work without a domain? (is Domain null or `""`)

Reply:
yeah, that should work, the dns daemon will just fail and never assign an entry